worker/src/global.rs

```diff
@@ -1,37 +1,75 @@
 use std::ops::Deref;
+use std::time::Duration;
 
+use reqwest::{Body, Client, Error, Response};
+use tokio::time;
 use wasm_bindgen::JsCast;
 use wasm_bindgen_futures::JsFuture;
 
 use crate::{
-    request::Request as WorkerRequest, response::Response as WorkerResponse, AbortSignal, Result,
+     AbortSignal, Result,
 };
 
 /// Construct a Fetch call from a URL string or a Request object. Call its `send` method to execute
 /// the request.
 pub enum Fetch {
     Url(url::Url),
-    Request(WorkerRequest),
+    Request(Body),
 }
 
 impl Fetch {
     /// Execute a Fetch call and receive a Response.
-    pub async fn send(&self) -> Result<WorkerResponse> {
+    pub async fn send(&self) -> Result<Response> {
         match self {
             Fetch::Url(url) => fetch_with_str(url.as_ref(), None).await,
             Fetch::Request(req) => fetch_with_request(req, None).await,
         }
     }
 
     /// Execute a Fetch call and receive a Response.
-    pub async fn send_with_signal(&self, signal: &AbortSignal) -> Result<WorkerResponse> {
+    pub async fn send_with_signal(&self, signal: &AbortSignal) -> Result<Response> {
```

Comment:
I wonder if we want to convert `reqwest::Response` to `http::Response<Body>`. I think its not totally straightforward, but possible.

Reply:
What would be the purpose of that? Would it help with keeping the data format more flexible? I'm curious.

Reply:
Well, right now if they wanted to call `fetch` and then return the value, it would be the wrong type unless it was converted to `http::Response`.